src/CMakeLists.txt

```diff
@@ -398,6 +398,9 @@ target_c_stack(swipl${PGO_SUFFIX} 4000000)
 if(MINGW)
   target_link_options(swipl${PGO_SUFFIX} PRIVATE -municode)
 endif()
+if(MSYS2)
+  add_compile_options(-DMSYS2)
```

Comment:
Probably more elegant to pass this through config.h.cmake.

Reply:
I see. Two more commits below.